crates/neon/src/types_impl/function/mod.rs

```diff
@@ -111,6 +175,109 @@ impl<'a> ConstructOptions<'a> {
     }
 }
 
+/// The trait for specifying values to be converted into arguments for a function call.
+/// This trait is sealed and cannot be implemented by types outside of the Neon crate.
+///
+/// **Note:** This trait is implemented for tuples of up to 32 JavaScript values,
+/// but for the sake of brevity, only tuples up to size 8 are shown in this documentation.
+pub trait TryIntoArguments<'cx>: private::TryIntoArgumentsInternal<'cx> {}
+
+impl<'cx> private::TryIntoArgumentsInternal<'cx> for () {
+    fn try_into_args_vec<C: Context<'cx>>(self, _cx: &mut C) -> NeonResult<private::ArgsVec<'cx>> {
+        Ok(smallvec![])
+    }
+}
+
+macro_rules! impl_into_arguments {
+    {
+        [ $(($tprefix:ident, $vprefix:ident), )* ];
+        [];
+    } => {};
+
+    {
+        [ $(($tprefix:ident, $vprefix:ident), )* ];
+        [ $(#[$attr1:meta])? ($tname1:ident, $vname1:ident), $($(#[$attrs:meta])? ($tnames:ident, $vnames:ident), )* ];
+    } => {
+        $(#[$attr1])?
+        impl<'cx, $($tprefix: TryIntoJs<'cx> + 'cx, )* $tname1: TryIntoJs<'cx> + 'cx> private::TryIntoArgumentsInternal<'cx> for ($($tprefix, )* $tname1, ) {
+            fn try_into_args_vec<C: Context<'cx>>(self, cx: &mut C) -> NeonResult<private::ArgsVec<'cx>> {
+                let ($($vprefix, )* $vname1, ) = self;
+                Ok(smallvec![ $($vprefix.try_into_js(cx)?.upcast(),)* $vname1.try_into_js(cx)?.upcast() ])
+             }
+         }
+
+         $(#[$attr1])?
+         impl<'cx, $($tprefix: TryIntoJs<'cx> + 'cx, )* $tname1: TryIntoJs<'cx> + 'cx> TryIntoArguments<'cx> for ($($tprefix, )* $tname1, ) {}
+
+         impl_into_arguments! {
+             [ $(($tprefix, $vprefix), )* ($tname1, $vname1), ];
+             [ $($(#[$attrs])? ($tnames, $vnames), )* ];
+         }
+     };
+}
+
+impl_into_arguments! {
```

Comment:
The [`from_args_impl`](https://github.com/neon-bindings/neon/blob/5d16c4d0c040c7d81565673ba4b42929922e03e1/crates/neon/src/types_impl/extract/mod.rs#L229-L288) macro could be good to look at to reduce the noise here a bit.